### PR TITLE
Declarative Configuration - Updates to technical details after breaking changes

### PR DESCRIPTION
**Pull request by kennsippell**

### CODE REVIEW

File: configuration/developing-community-health-applications.md [hunk 1]
```diff
@@ -520,125 +529,122 @@ To build your targets into your app, you must compile them into app-settings, th
...
-| `fields.width` | The horizontal space for the field. Common values are 12 for full width, 6 for half width, or 3 for quarter width.| |
-| `fields.appliesIf` | A function that determines when the field should be shown. | no |
-| `fields.translate` | Whether or not to translate the value. Defaults to false. <!--TODO: VERIFY IMPLEMENTED--> | no |
-| `fields.context` |  The fields available in the value's translation. eg {} | no | 
-| `fields.icon` | The name of the icon to display beside this field, as defined through the Configuration > Icons page. | no |
-| `modifyContext` | function(ctx) | Used to modify or add values of the `context` | no |
+| property | type | description | required |
+|---|---|---|--|
+| `label` | `translation key` | Label on top of card | yes |
+| `fields` | Array of `Fields` | The content of the card | yes |
```
Comment (abbyad) [inline configuration/developing-community-health-applications.md]: I like the idea of consolidating to a common Field type. 

Are we sure that all fields in `Fields` are valid here? Eg `appliesToType`

Also, some available in this schema are not in `Fields` eg `fields.context`.

If we can reconcile those difference this makes a lot of sense.

File: configuration/developing-community-health-applications.md [hunk 2]
```diff
@@ -520,125 +529,122 @@ To build your targets into your app, you must compile them into app-settings, th
     medic-conf --local compile-app-settings backup-app-settings upload-app-settings 
 
 
-## Contacts
-In the `app_settings.contact_summary` you can write a script to output fields for the contact info pane and help decide which reports should be able to be filed against a contact. The script is evaluated as JavaScript so all the standard language features are available. To make this easier to write and maintain, a declarative format is also available. In the `contact_summary.templated.js` file you specify three variables: `context`, `fields`, and `cards`.
+## Contact Summaries
```
Comment (abbyad) [inline configuration/developing-community-health-applications.md]: Even though this code is stored in `app_settings.contact_summary`, and the file for it is `contact-summary.templated.js`, I don't think _Contact Summaries_ defines this section well because it affects the context for the actions as well. What about _Contact Pages_, or _Contact Profiles_?

We need to remember to update the index as well if we change this from _Contacts_.

File: configuration/developing-community-health-applications.md [hunk 3]
```diff
@@ -520,125 +529,122 @@ To build your targets into your app, you must compile them into app-settings, th
...
+In the `contact-summary.templated.js` file, write a script to control how a contact's profile page will appear by defining three outputs: `context`, `fields`, and `cards`.
```
Comment (abbyad) [inline configuration/developing-community-health-applications.md]: We should add an intro about Contact Pages, like we do for Tasks and Targets. Also, I think we should shift away from "write a script" since we are describing the declarative format in the sections below, even if we don't say it directly.

```suggestion
Health workers can easily access and manage the families and people they serve as profile pages in the People tab. These contact profiles display basic information about the contact along with their history and upcoming tasks. You can control how a contact's profile page will appear and behave by defining these three outputs in the `contact-summary.templated.js` file: `context`, `fields`, and `cards`.
```

File: configuration/developing-community-health-applications.md [hunk 4]
```diff
@@ -141,34 +141,40 @@ The meta information in the `{form_name}.properties.json` file defines the form'
...
+      "expression": "contact.type === 'person' && (!contact.sex || contact.sex === 'female') && (!contact.date_of_birth || (ageInYears(contact) >= 10 && ageInYears(contact) < 65))"
     }
+  }
+```
 
 #### Build
     
 Forms are converted with the `convert-app-forms` or `convert-contact-forms` actions in `medic-conf`, and added to your application with the corresponding upload actions.
 
+    medic-conf --local convert-app-forms upload-app-form
```
Comment (abbyad) [inline configuration/developing-community-health-applications.md]:
```suggestion
    medic-conf --local convert-app-forms upload-app-forms
```

File: configuration/developing-community-health-applications.md [hunk 5]
```diff
@@ -141,34 +141,40 @@ The meta information in the `{form_name}.properties.json` file defines the form'
...
+      "person": true,
+      "place": false,
+      "expression": "contact.type === 'person' && (!contact.sex || contact.sex === 'female') && (!contact.date_of_birth || (ageInYears(contact) >= 10 && ageInYears(contact) < 65))"
     }
+  }
+```
 
 #### Build
     
 Forms are converted with the `convert-app-forms` or `convert-contact-forms` actions in `medic-conf`, and added to your application with the corresponding upload actions.
```
Comment (abbyad) [inline configuration/developing-community-health-applications.md]: Just cleaning up while we are here for consistency.

```suggestion
Convert and build the app forms into your application using the `convert-app-forms` and `upload-app-forms` actions in `medic-conf`.
```

File: configuration/developing-community-health-applications.md [hunk 6]
```diff
@@ -184,63 +190,63 @@ Previously, the code to generate tasks would iterate through an object with all
...
+| `events` | Array of events | An event is used to specify the timing of the task. | yes |
+| `events[n].id` | string | Can help as a descriptive name (eg `pregnancy-high-risk`). One task will appear per unique id, so re-using ids can be useful to avoid duplicate tasks appearing. | no |
+| `events[n].days` | int | Number of days after the doc's `reported_date` that the event is due | yes, if `dueDate` is not set |
+| `events[n].dueDate` | `function(event, contact, report)` | Returns a `Date` object for the day when this event is due. | yes, if `days` is not set |
+| `events[n].start` | int | Number of days to show the task before it is due | yes |
+| `events[n].end` | int | Number of days to show the task after it is due | yes |
+| `actions` | Array of actions | The actions (forms) that a user can access after clicking on a task. If you put multiple forms here, the user will see a task summary screen where they can select which action they would like to complete. | yes |
+| `actions[n].type` | `'report'` or `'contact'` | When `'report'`, the action opens the given form. When `'contact'`, the action redirects to a contact's profile page. Defaults to 'report'. | no |
```
[the remark this answers is not in the document]
Comment (kennsippell) [reply configuration/developing-community-health-applications.md]: I looked up the code and I'm pretty sure this is what is happening. I didn't tested it though. We could remove it if nobody ever uses it?

File: configuration/developing-community-health-applications.md (hunk 2 above)
Comment (kennsippell) [reply configuration/developing-community-health-applications.md]: I'm fine just reverting back to Contacts.

File: configuration/developing-community-health-applications.md (hunk 1 above)
Comment (kennsippell) [reply configuration/developing-community-health-applications.md]: You're right. Some weird stuff happening...

Comment (kennsippell) [reply configuration/developing-community-health-applications.md]: Okay - reverted and corrected.

File: configuration/developing-community-health-applications.md [hunk 7]
```diff
@@ -548,13 +548,14 @@ Each field that can be shown on a contact's profile is defined as an object in t
...
+| `icon` | `string` | The name of the icon to display beside this field, as defined through the Configuration > Icons page. | no |
+| `value` | `string` | The value shown for the field | yes |
+| `filter` | `string` | The display filter to apply to the value, eg: `{ value: '2005-10-09', filter: 'age' }` will render as "11 years". Common filters are: `age`, `phone`, `weeksPregnant`, `relativeDate`, `relativeDay`, `fullDate`, `simpleDate`, `simpleDateTime`, `lineage`, `resourceIcon`. For the complete list of filters, and more details on what each does, check out the code in [`medic/webapp/src/js/filters` dir](https://github.com/medic/medic/tree/master/webapp/src/js/filters). | no |
+| `width` | `integer` | The horizontal space for the field. Common values are 12 for full width, 6 for half width, or 3 for quarter width. Default 12 | no |
+| `translate` | `boolean` | Whether or not to translate the value. Defaults to false. | no |
+| `context` | `object` | [Translation variables](https://angular-translate.github.io/docs/#/guide/06_variable-replacement) | no |
```
Comment (abbyad) [inline configuration/developing-community-health-applications.md]: The word _context_ is used in many different ways across this documentation, and it is not clear what this `context` field does. We should add some ~context~ info to this `context` field, to make it clear whether or not someone would want to use this without going to the external documentation.

Comment (abbyad) [reply configuration/developing-community-health-applications.md]: An example would be helpful for this too.

File: configuration/developing-community-health-applications.md [hunk 8]
```diff
@@ -548,13 +548,14 @@ Each field that can be shown on a contact's profile is defined as an object in t
...
+| `appliesIf` | `boolean` | True if the field should be shown. | no |
```
Comment (abbyad) [inline configuration/developing-community-health-applications.md]: This one hasn't changed as part of this PR, but is missing the function prototype. Worth adding while we are here.

File: configuration/developing-community-health-applications.md [hunk 9]
```diff
@@ -565,10 +566,18 @@ An array of cards to show below the summary on the profile page. Each card has i
...
 | `label` | `translation key` | Label on top of card | yes |
-| `fields` | Array of `Fields` | The content of the card | yes |
-| `appliesToType` | Array of contact types | Only calls `appliesIf` if the contact is a match. For example, `['person']`. | no |
-| `appliesIf` | boolean | True if the field should be shown. | no |
-| `modifyContext` | `function(context)` | Used to modify or add values of the `context` | no |
+| `appliesToType` | Array of contact types `string` | Only calls `appliesIf` if the contact's type matches one or more of the array elements. For example, `['person']`. | no |
+| `appliesIf` | `boolean` | True if the field should be shown. | no |
+| `modifyContext` | `function(context)` | Used to modify or add values of the data which is passed as input to forms filled from the contact page | no |
+| `fields` | `Array[]` of [fields](#fields) | The content of the card | yes |
+| `fields[n].appliesIf` | `boolean` or `function(report)` | Same as Fields.appliesIf above | |
```
Comment (abbyad) [inline configuration/developing-community-health-applications.md]: When reading the markdown as a reference it is hard to follow by jumping back and forth with the Fields section. I would duplicate the info, and we can still note that this is the same as Fields.appliesIf above. Listing the required fields here is also useful without having to jump around in the doc.

File: configuration/developing-community-health-applications.md (hunk 7 above)
Comment (kennsippell) [reply configuration/developing-community-health-applications.md]: Ya. I found that really confusing. I previously opened this https://github.com/medic/medic-conf/issues/169

**Comment by kennsippell** [comment]
Holding off on merging this until the associated `medic-conf` changes are merged and we're ready to publish everything via npm.
